diff --git a/Response/CurlResponse.php b/Response/CurlResponse.php
index 7e941c6..c80cace 100644
--- a/Response/CurlResponse.php
+++ b/Response/CurlResponse.php
@@ -145,19 +145,18 @@ final class CurlResponse implements ResponseInterface, StreamableInterface
             });
         }
 
-        curl_setopt($ch, \CURLOPT_WRITEFUNCTION, static function ($ch, string $data) use ($multi, $id): int {
-            if ('H' === (curl_getinfo($ch, \CURLINFO_PRIVATE)[0] ?? null)) {
-                $multi->handlesActivity[$id][] = null;
-                $multi->handlesActivity[$id][] = new TransportException(\sprintf('Unsupported protocol for "%s"', curl_getinfo($ch, \CURLINFO_EFFECTIVE_URL)));
+        $checkProtocol = true;
+        curl_setopt($ch, \CURLOPT_WRITEFUNCTION, static function ($ch, string $data) use ($multi, $id, &$checkProtocol): int {
+            if ($checkProtocol) {
+                if ('H' === (curl_getinfo($ch, \CURLINFO_PRIVATE)[0] ?? null)) {
+                    $multi->handlesActivity[$id][] = null;
+                    $multi->handlesActivity[$id][] = new TransportException(\sprintf('Unsupported protocol for "%s"', curl_getinfo($ch, \CURLINFO_EFFECTIVE_URL)));
 
-                return 0;
-            }
-
-            curl_setopt($ch, \CURLOPT_WRITEFUNCTION, static function ($ch, string $data) use ($multi, $id): int {
-                $multi->handlesActivity[$id][] = $data;
+                    return 0;
+                }
 
-                return \strlen($data);
-            });
+                $checkProtocol = false;
+            }
 
             $multi->handlesActivity[$id][] = $data;
