--- a/src/Test/IntegrationTestCase.php	2026-06-27 17:17:21.445146156 +0200
+++ b/src/Test/IntegrationTestCase.php	2026-06-27 17:17:40.805108949 +0200
@@ -144,13 +144,13 @@
      *
      * @final since Twig 3.13
      */
-    public function getTests($name, $legacyTests = false)
+    public static function getTests($name, $legacyTests = false)
     {
         try {
             $fixturesDir = static::getFixturesDirectory();
         } catch (\BadMethodCallException) {
             trigger_deprecation('twig/twig', '3.13', 'Not overriding "%s::getFixturesDirectory()" in "%s" is deprecated. This method will be abstract in 4.0.', self::class, static::class);
-            $fixturesDir = $this->getFixturesDir();
+            $fixturesDir = (new static())->getFixturesDir();
         }
 
         $fixturesDir = realpath($fixturesDir);
@@ -201,9 +201,9 @@
      *
      * @return iterable
      */
-    public function getLegacyTests()
+    public static function getLegacyTests()
     {
-        return $this->getTests('testLegacyIntegration', true);
+        return static::getTests('testLegacyIntegration', true);
     }
 
     /**
