diff --git a/app/bundles/PageBundle/EventListener/PageSubscriber.php b/app/bundles/PageBundle/EventListener/PageSubscriber.php
index 9197bb980..2f704aa5a 100644
--- a/app/bundles/PageBundle/EventListener/PageSubscriber.php
+++ b/app/bundles/PageBundle/EventListener/PageSubscriber.php
@@ -21,6 +21,7 @@ use Mautic\PageBundle\PageEvents;
 use Mautic\QueueBundle\Event\QueueConsumerEvent;
 use Mautic\QueueBundle\Queue\QueueConsumerResults;
 use Mautic\QueueBundle\QueueEvents;
+use Mautic\PageBundle\Entity\Page;
 
 /**
  * Class PageSubscriber.
@@ -189,7 +190,6 @@ class PageSubscriber extends CommonSubscriber
         $hitId                  = $payload['hitId'];
         $pageId                 = $payload['pageId'];
         $leadId                 = $payload['leadId'];
-        $isRedirect             = !empty($payload['isRedirect']);
         $hitRepo                = $this->em->getRepository('MauticPageBundle:Hit');
         $pageRepo               = $this->em->getRepository('MauticPageBundle:Page');
         $redirectRepo           = $this->em->getRepository('MauticPageBundle:Redirect');
@@ -197,6 +197,8 @@ class PageSubscriber extends CommonSubscriber
         $hit                    = $hitId ? $hitRepo->find((int) $hitId) : null;
         $lead                   = $leadId ? $leadRepo->find((int) $leadId) : null;
 
+        $page = NULL;
+
         // On the off chance that the queue contains a message which does not
         // reference a valid Hit or Lead, discard it to avoid clogging the queue.
         if (null === $hit || null === $lead) {
@@ -213,10 +215,15 @@ class PageSubscriber extends CommonSubscriber
             return;
         }
 
-        if ($isRedirect) {
-            $page = $pageId ? $redirectRepo->find((int) $pageId) : null;
-        } else {
-            $page = $pageId ? $pageRepo->find((int) $pageId) : null;
+        if ($pageId) {
+            $foundPage = $pageRepo->find((int) $pageId);
+
+            if ($foundPage instanceof Page) {
+              $page = $foundPage;
+            }
+            else {
+              $page = $redirectRepo->find((int) $pageId);
+            }
         }
 
         // Also reject messages when processing causes any other exception.
