diff --git a/node_modules/at.js/dist/js/jquery.atwho.js b/node_modules/at.js/dist/js/jquery.atwho.js
index 09ca40c..dfed389 100644
--- a/node_modules/at.js/dist/js/jquery.atwho.js
+++ b/node_modules/at.js/dist/js/jquery.atwho.js
@@ -835,13 +835,13 @@ EditableController = (function(superClass) {
     }
     suffix = (suffix = this.getOpt('suffix')) === "" ? suffix : suffix || "\u00A0";
     data = $li.data('item-data');
-    this.query.el.removeClass('atwho-query').addClass('atwho-inserted').html(content).attr('data-atwho-at-query', "" + data['atwho-at'] + this.query.text).attr('contenteditable', "false");
+      this.query.el.removeClass('atwho-query').addClass('atwho-inserted').html(content).attr('data-atwho-at-query', "" + data['atwho-at'] + this.query.text);
     if (range = this._getRange()) {
       if (this.query.el.length) {
         range.setEndAfter(this.query.el[0]);
       }
       range.collapse(false);
-      range.insertNode(suffixNode = this.app.document.createTextNode("" + suffix));
+        range.insertNode(suffixNode = this.app.document.createTextNode("\u200D" + suffix));
       this._setRange('after', suffixNode, range);
     }
     if (!this.$inputor.is(':focus')) {
diff --git a/node_modules/at.js/src/editableController.coffee b/node_modules/at.js/src/editableController.coffee
index d6fcaeb..5f76987 100644
--- a/node_modules/at.js/src/editableController.coffee
+++ b/node_modules/at.js/src/editableController.coffee
@@ -163,12 +163,11 @@ class EditableController extends Controller
       .addClass 'atwho-inserted'
       .html content
       .attr 'data-atwho-at-query', "" + data['atwho-at'] + @query.text
-      .attr 'contenteditable', "false"
     if range = @_getRange()
       if @query.el.length
         range.setEndAfter @query.el[0]
       range.collapse false
-      range.insertNode suffixNode = @app.document.createTextNode "" + suffix
+      range.insertNode suffixNode = @app.document.createTextNode "\u200D" + suffix
       @_setRange 'after', suffixNode, range
     @$inputor.focus() unless @$inputor.is ':focus'
     @$inputor.change()
